### PR TITLE
refactor(docs): better styling for docs

### PR DESCRIPTION
**Pull request by asrvd**
## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

[WIP]
- changed the icons for copy button
- reduced border radius for code blocks
- added little bottom margin to `h3` elements 
- border color for code blocks (inspired by astro docs)
- darker code block bg-color
- darker blockquote color for trpc 
- smaller font sizes

---

## Screenshots

1. added bg color to nav elements and reduced font sizes

before:
image
after:
image

2. changed bg-color for code blocks and added better icons, reduced border-radius and added some fancy border

before:
image
after:
image

3. changed bg-color of blockquote in trpc and also made the name non-italic, removed unnecessary margin

before: 
image
after:
image

💯

### CODE REVIEW

**Comment by asrvd** [comment]
don't merge!

**Comment by nexxeln** [comment]
> don't merge!

make it a draft

**Comment by c-ehrlich** [comment]
When you are ready for review, could you give some before/after screenshots of the specific things you changed? With styling PRs its sometimes hard to figure out what is actually different.

**Comment by asrvd** [comment]
> When you are ready for review, could you give some before/after screenshots of the specific things you changed? With styling PRs its sometimes hard to figure out what is actually different.

sure

**Comment by asrvd** [comment]
@c-ehrlich i have updated the pr with some images, you may have a look

**Review by c-ehrlich — COMMENTED** [review]
Reviewing how it looks here, not the code itself:

stuff i love:
- smaller font sizes: i like this a lot. the proportions of everything feel better than before.
- bottom margin for h3: this is great, proportions on pages with them look much better
- new copy button icon: much better than before, the old one looks like a paste button

stuff like some parts of:
- quote element
  - i think its an improvement overall
  - small font is better than before
  - i prefer the old background color in dark mode, because it gives more contrast against the background. if anything, i'd increase the contrast in light mode as well
  - not a fan of the drop shadow because we're not using it anywhere else
  - it can still use a lot of work unrelated to the changes in this pr: for example the font looks a bit off, and the avatar and name feel too close to the edge
- code blocks
  - i find this version easier to read than the old one, which is probably the most important thing about code blocks
  - the border looks ok in dark mode but i don't like it in light mode because it blends in with the background and then you only see the divider
  - I think the Remix docs have one of the nicest code block elements that I've ever seen
image
The Next.js beta docs have a great code block also. Maybe we can learn something from one of them? They do have it easier though because they're both set against a nearly-black background


stuff i don't like:
- nav element background
  - i do agree that the active element should stand out a bit more, but maybe this can be achieved with more color contrast and/or more weight?

File: www/src/components/docs/pageContent.astro [hunk 1]
```diff
@@ -73,25 +57,7 @@ const title = frontmatter.title;
...
-    <path
-      d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
-      stroke="none"
-    />
-    <path
-      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
-      stroke="none"
-    />
-  </svg>`;
+        copyButton.innerHTML = `<svg width="1.3em" height="1.3em" viewBox="0 0 256 256"><path fill="currentColor" d="M104 192a8.5 8.5 0 0 1-5.7-2.3l-56-56a8.1 8.1 0 0 1 11.4-11.4l50.3 50.4L210.3 66.3a8.1 8.1 0 0 1 11.4 11.4l-112 112a8.5 8.5 0 0 1-5.7 2.3Z"/></svg>`;
```
Comment (juliusmarminge) [inline www/src/components/docs/pageContent.astro:60]: Green checkmark?

File: www/src/styles/global.css [hunk 2]
```diff
@@ -159,26 +159,23 @@
...
+    @apply mx-auto mb-3 rounded border border-t3-purple-300/80 shadow-xl p-2 pl-3 font-mono sm:pt-2;
   }
 
   .markdown ol {
     @apply w-full;
   }
 
   .markdown details {
-    @apply mt-2 w-full rounded-md  bg-[#292D3E] px-3 transition-all duration-75 ease-in-out;
+    @apply mt-2 w-full rounded-md  bg-[#0F111A] px-3 transition-all duration-75 ease-in-out;
```
Comment (juliusmarminge) [inline www/src/styles/global.css:174]: These look a bit weird with the border on the inner element?

image

**Review by juliusmarminge — COMMENTED** [review]
I like the bg on the active sidebar-item.

CodeBlocks are kinda hard to read I think (or maybe it's just cause it's late xD). They would require a complete revamp I think like @c-ehrlich mentioned.

File: www/src/styles/global.css (hunk 2 above)
Comment (asrvd) [reply www/src/styles/global.css:174]: yep, even I don;t like it in the details block, is there any way we can filter it out from details block

Comment (asrvd) [reply www/src/styles/global.css:174]: ig we can by using css, something like `.details > pre` maybe

**Comment by asrvd** [comment]
got most of the things done, will be leaving the codeblock theme as `rose-pine` for now and will add our own custom theme later

**Review by nexxeln — COMMENTED** [review]
Codeblocks have cursor-pointer. It should be normal cursor

**Comment by nexxeln** [comment]
Looks like we can merge now. Did you have some font suggestions?

**Comment by nexxeln** [comment]
> also, maybe add line numbers to code blocks 🤔

eh who cares

**Comment by c-ehrlich** [comment]
> > also, maybe add line numbers to code blocks 🤔
> 
> eh who cares

yea agreed, doesn't really matter for our code blocks

**Comment by asrvd** [comment]
alright

**Comment by asrvd** [comment]
btw need opinions on this, I dont really like scrollbars floating like this
image

**Comment by nexxeln** [comment]
It's fine imo

**Comment by asrvd** [comment]
anything else we need in this pr or can this get merged now?

**Review by nexxeln — APPROVED** [review]
Everything looks good to me

**Comment by juliusmarminge** [comment]
You've missed styling of the Theo quote: 
CleanShot 2022-11-15 at 23 28 59

File: www/src/styles/global.css [hunk 3]
```diff
@@ -159,26 +159,29 @@
...
+    @apply mx-auto mb-3 rounded border border-t3-purple-300/80 p-2 pl-3 font-mono shadow-xl sm:pt-2;
   }
 
   .markdown ol {
     @apply w-full;
   }
 
   .markdown details {
-    @apply mt-2 w-full rounded-md  bg-[#292D3E] px-3 transition-all duration-75 ease-in-out;
+    @apply mt-2 w-full rounded-md bg-[#191724] border border-t3-purple-300/80 px-3 transition-all duration-75 ease-in-out;
```
Comment (juliusmarminge) [inline www/src/styles/global.css:174]:
```suggestion
    @apply mt-2 w-full rounded-md bg-[#191724] border border-t3-purple-300/80 px-3 transition-all duration-300 ease-in-out;
```

Comment (c-ehrlich) [reply www/src/styles/global.css:174]: should the quote maybe be a component? makes it easier to keep them consistent, especially if we were to add more of them in the future

Comment (juliusmarminge) [reply www/src/styles/global.css]: probably

**Review by juliusmarminge — APPROVED** [review]
LGTM

**Pull request merged**